dotCMS/src/main/java/com/dotmarketing/servlets/LoginEditModeServlet.java

```diff
@@ -43,15 +45,23 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
 
 					if ("contentlet".equals(_edit_mode_id.getAssetType())) {
 						com.dotmarketing.portlets.contentlet.model.Contentlet cont = APILocator.getContentletAPI().findContentletByIdentifier(_edit_mode_id.getId(), false, APILocator.getLanguageAPI().getDefaultLanguage().getId(), APILocator.getUserAPI().getSystemUser(), false);
-						_edit_mode_id.setURI(cont.getMap().get("URL_MAP_FOR_CONTENT").toString());
+						String pageURI = "";
+						if(UtilMethods.isSet(cont.getMap().get("URL_MAP_FOR_CONTENT")))
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00121)